include/fmt/printf.h

```diff
@@ -212,14 +212,12 @@ class printf_arg_formatter
     : public internal::function<
           typename internal::arg_formatter_base<Range>::iterator>,
       public internal::arg_formatter_base<Range> {
- private:
+ public:
```

Comment:
I don't think these should be public because the children won't be able to use these types directly anyway.

Reply:
Agreed. But I have to keep the iterator public (like in the arg_formatter) to get my approach working.

Reply:
Right, making `iterator` public makes sense.